lib/shared.bash

```diff
@@ -102,14 +102,30 @@ function build_image_override_file_with_version() {
     exit 1
   fi
 
+  cache_from_not_available=($(bc <<< "$version < 3.2"))
```

Comment:
Unfortunately, I suspect we will have to do specific docker-compose version checks, rather than just the file version.

Reply:
as per https://docs.docker.com/compose/compose-file/ I think this is the compose file version
```
CACHE_FROM

> Note: This option is new in v3.2
```

Reply:
Rather than bring in `bc`, you can use bash arithmatic contexts:

```
if (( a > b )); then
    ...
fi
```

Reply:
Unfortunately it was broken until docker-compose 1.14.0 for multiple files, but at least it didn't fail with an error. This check is probably ok.

Reply:
as far as I know bash arithmetics does not work with floats. I would have to either multiply it by something to make it integer or split it by the dot and compare parts manually. I'm not that good at bash though so maybe there's another way